src/System.Management.Automation/engine/ParameterBinderBase.cs

```diff
@@ -576,7 +576,15 @@ internal CommandLineParameters CommandLineParameters
 
                     try
                     {
-                        BindParameter(parameter.ParameterName, parameterValue, parameterMetadata);
+                        // After transformations and validations, if the parameter value is still null,
+                        // if it was originally a null literal, and if the parameter has explicit support
+                        // for a null literal, bind the NullLiteral singleton instead.
+                        BindParameter(
+                            parameter.ParameterName,
+                            parameterValue == null && parameter.ArgumentValue == null && parameter.ArgumentNullLiteral && parameterMetadata.SupportsNullLiteralArgument
```

Comment:
It doesn't seem like the first 2 tests are necessary. I suppose you could add an assertion, but if a converted value is not null, it seems like either you shouldn't have specified `SupportsNullLiteralArgument` or you really want the `NullLiteral.Value` instead of the converted value.

Reply:
The second test is definitely not necessary, because `parameter.ArgumentNullLiteral` covers that scenario, so I removed that test in the next commit.

I think it's worth keeping the first test there though because the parameter goes through transformation attributes before it reaches this code, and I wanted to make sure that everything still worked in that case. Sure, it means that you could specify `SupportsNullLiteralArgument` and then transform `null` into something non-null, in which case the `SupportsNullLiteralArgument` attribute doesn't do anything, but that seems like a command design issue for a tool such as PowerShell Script Analyzer to catch. Better for the code to still work in that unusual scenario than to assert, in my opinion.